File: csharp/triangle/Triangle.cs
using System;
using System.Linq;

public enum TriangleKind
{
    Equilateral,
    Isosceles,
    Scalene
}

public static class Triangle
{
    public static TriangleKind Kind(decimal side1, decimal side2, decimal side3)
    {
        var sides = new[] { side1, side2, side3 }.OrderBy(i => i).ToArray();

        if( sides.Any(i=> i <= 0))
        {
            throw new TriangleException("A sides must be greater than 0 in length");
        }

        if (sides[0] + sides[1] <= sides[2])
        {
            throw new TriangleException("The triangle rule is violated");
        }

        TriangleKind kind;
        if (sides[0] == sides[1] && sides[1] == sides[2])
        {
            kind = TriangleKind.Equilateral;
        }
        else if (sides[0] == sides[1] || sides[1] == sides[2])
        {
            kind = TriangleKind.Isosceles;
        }
        else
        {
            kind = TriangleKind.Scalene;
        }

        return kind;
    }
}

public class TriangleException : Exception {
    public TriangleException()
    {
    }

    public TriangleException(string message) : base(message)
    {
        
    }
}

File: python/triangle/triangle.py
"""exercise"""

def is_triange(test):
    """
    :param sides: length of sides
    :return: True if a triangle; otherwise, false
    """
    return lambda sides: all(length > 0 for length in sides) and 2*max(sides) < sum(sides) and test(sides)


@is_triange
def equilateral(sides):
    """
    :param sides: length of sides
    """
    return (sides[0] == sides[1]) and (sides[1] == sides[2])


@is_triange
def isosceles(sides):
    """
    :param sides: length of sides
    """
    return ((sides[0] == sides[1]) or (sides[1] == sides[2]) or (sides[0] == sides[2]))


@is_triange
def scalene(sides):
    """
    :param sides: length of sides
    """
    return sides[0] != sides[1] and sides[1] != sides[2] and sides[0] != sides[2]
